components/tidb_query_datatype/src/codec/data_type/bit_vec.rs

```diff
@@ -1,17 +1,17 @@
 // Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
 
-/// A boolean vector, which consolidates 8 booleans into 1 u8 to save space.
+/// A boolean vector, which consolidates 64 booleans into 1 u64 to save space.
 ///
 /// `BitVec` is mainly used to implement bitmap in ChunkedVec.
 #[derive(Debug, PartialEq, Clone)]
 pub struct BitVec {
```

Comment:
Why not use https://docs.rs/bitvec/0.19.2/bitvec/vec/struct.BitVec.html?

Reply:
When evaluating vectorized functions, we need to apply bitwise and on several BitVec. This can be done efficiently by bitwise and a whole u8 or u64. When I was prototyping the original version of BitVec, it seems that bitvec crate doesn't provide such functionality.

Reply:
Seems that there is a [bitwise and function](https://docs.rs/bitvec/0.19.2/bitvec/vec/struct.BitVec.html#method.bitand) for bitvec crate. Maybe we could migrate to `bitvec` and do some benchmark.

Reply:
I ran the micro benchmarks of `ChunkedVecSized` by removing the `push` into `BitVec` and what I found from the results on my machine is that `BitVec` is not the biggest contributor to overhead. Do you think it's still worth exploring `bitvec` crate?

Reply:
From my benchmark result https://github.com/tikv/tikv/issues/8504 it seems that BitVec push takes 1.5x time of vec push, that's why I suggest optimize bitvec performance. I may run the micro benchmark later to verify the result.